src/libraries/Native/AnyOS/System.IO.Compression.Native/entrypoints.c

```diff
@@ -0,0 +1,42 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "../../AnyOS/entrypoints.h"
+
+// Include System.IO.Compression.Native headers
+#include "../zlib/pal_zlib.h"
+#include "../brotli/include/brotli/decode.h"
+#include "../brotli/include/brotli/encode.h"
+#include "../brotli/include/brotli/port.h"
+#include "../brotli/include/brotli/types.h"
+
+#include "../../AnyOS/entrypoints.h"
+
+static const Entry s_compressionNative[] =
+{
+    DllImportEntry(BrotliDecoderCreateInstance)
+    DllImportEntry(BrotliDecoderDecompress)
+    DllImportEntry(BrotliDecoderDecompressStream)
+    DllImportEntry(BrotliDecoderDestroyInstance)
+    DllImportEntry(BrotliDecoderIsFinished)
+    DllImportEntry(BrotliEncoderCompress)
+    DllImportEntry(BrotliEncoderCompressStream)
+    DllImportEntry(BrotliEncoderCreateInstance)
+    DllImportEntry(BrotliEncoderDestroyInstance)
+    DllImportEntry(BrotliEncoderHasMoreOutput)
+    DllImportEntry(BrotliEncoderSetParameter)
+    DllImportEntry(CompressionNative_Crc32)
+    DllImportEntry(CompressionNative_Deflate)
+    DllImportEntry(CompressionNative_DeflateEnd)
+    DllImportEntry(CompressionNative_DeflateInit2_)
+    DllImportEntry(CompressionNative_Inflate)
+    DllImportEntry(CompressionNative_InflateEnd)
+    DllImportEntry(CompressionNative_InflateInit2_)
+};
+
+EXTERN_C const void* CompressionResolveDllImport(const char* name);
```

Comment:
A nit - this line is not necessary.

Reply:
Did you mean the whitespace or the forward declaration?

Reply:
The prototype is needed to make GCC happy about extern.


/runtime/src/libraries/Native/Unix/System.Globalization.Native/entrypoints.c(58,22): error G8EE6BEF4: no previous prototype for function 'GlobalizationResolveDllImport' [-Werror,-Wmissing-prototypes] [/runtime/src/libraries/Native/build-native.proj]


I have seen such seemingly redundant declarations in other places too.

Reply:
yea `-Wmissing-prototypes` wants to see a separate declaration for each non-`static` function definition.  (normally you'd have these in a `.h` somewhere that you then include in your `.c` or `.cpp` file)

Reply:
Ah, I am too much influenced by C++, I haven't used C for a very long time. Thank you for the explanation.